server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -501,8 +511,47 @@ private static SnapshotsInRepo sortSnapshots(
         final @Nullable GetSnapshotsRequest.After after,
         final int offset,
         final int size,
-        final SortOrder order
+        final SortOrder order,
+        final String[] slmPolicies
     ) {
+        final List<SnapshotInfo> filteredSnapshotInfos;
+        if (slmPolicies.length == 0) {
+            filteredSnapshotInfos = snapshotInfos;
+        } else {
+            final List<String> includePatterns = new ArrayList<>();
+            final List<String> excludePatterns = new ArrayList<>();
+            boolean seenWildcard = false;
+            for (String slmPolicy : slmPolicies) {
+                if (seenWildcard && slmPolicy.length() > 1 && slmPolicy.startsWith("-")) {
+                    excludePatterns.add(slmPolicy.substring(1));
+                } else {
+                    if (Regex.isSimpleMatchPattern(slmPolicy)) {
+                        seenWildcard = true;
+                    }
+                    includePatterns.add(slmPolicy);
+                }
+            }
+            final String[] includes = includePatterns.toArray(Strings.EMPTY_ARRAY);
+            final String[] excludes = excludePatterns.toArray(Strings.EMPTY_ARRAY);
+            filteredSnapshotInfos = snapshotInfos.stream().filter(snapshotInfo -> {
+                final Map<String, Object> metadata = snapshotInfo.userMetadata();
+                final String policy;
```

Comment:
I was wondering if were possible to extract getting policy from the metadata to own method in order to make policy `final`.

```java
String getPolicy(Metadata metadata) {
  if (metadata != null) {
    final Object policyFound = metadata.get(SnapshotsService.POLICY_ID_METADATA_FIELD);
    if (policyFound instanceof String) {
       return (String) policyFound;
    }
  } 
  return "";
}
```

Reply:
Makes sense, I'll extract this logic in a follow-up then we can dry it up with SLM code in one go :)